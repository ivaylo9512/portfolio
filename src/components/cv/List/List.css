.list{
    margin-top: 5%;
    padding-right: 2%;
}