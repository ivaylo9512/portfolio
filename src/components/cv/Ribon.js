const Ribon = ({ name, color }) => {
    return(
        <div className={color}>
            {name}
        </div>
    )
}

export default Ribon