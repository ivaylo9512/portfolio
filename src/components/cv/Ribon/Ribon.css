.ribon{
    height: 3.3vw;
    background: black;
    border-radius: 0.3vw;
}