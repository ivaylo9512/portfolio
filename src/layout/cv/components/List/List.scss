.list{
    font-size: 1.5vw;
    width: 70%;
    margin-bottom: 4vh;

    > div:first-child{
        fill: var(--clr-primary);
    }
    
    > div:not(:first-child){
        text-transform: uppercase;
        margin-top: 3.2%;
        fill: white;
    }
    
    > :first-child .name{
        height: 50%;
    }

    @media only screen and (min-width: 700px) and (orientation: landscape) {
        width: 100%;
        margin-bottom: 2vh;
    }
}
.name{
    fill: black;

    path{
        filter: url(#name-shadow);
    }
}

.list-svg{
    height: 50%;
    fill: black;
    margin-left: auto;
    margin-right: 6%;
}