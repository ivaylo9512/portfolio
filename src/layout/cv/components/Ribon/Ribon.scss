.ribon{
    position: relative;
    opacity: 0.9;

    > svg{
        display: block;
    }

    rect{
        filter: url(#shape-shadow);
    }

    .content{
        position: absolute;
        display: flex;
        width: 100%;
        height: 100%;
        padding-left: 3%;
        z-index: 5;
        box-sizing: border-box;
        align-items: center;
    }

    @media only screen and (min-width: 700px) {
        opacity: 1; 
    }
}