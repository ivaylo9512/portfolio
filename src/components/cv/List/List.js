import Ribon from '../Ribon/Ribon';
import './List.css';

const List = ({ name, children }) => {
    return(
        <div className='list'> 
            <Ribon name={name}/>
            {children}
        </div>
    )
}

export default List