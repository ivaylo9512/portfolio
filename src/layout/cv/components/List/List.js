import Ribon from '@cv/components/Ribon/Ribon';
import ListSvg from '@svgIcons/ListSvg';
import './List.scss';
import { useEffect, useRef, useState } from 'react';
import useResize from 'AppRoot/hooks/useResize';

const List = ({ ribon, children, className }) => {
    const overflow = useRef();
    const {dimensions} = useResize();
    const [height, setHeight] = useState();
    const [toggle, setToggle] = useState(false);
    const [isAnimating, setIsAnimating] = useState(false);
    const [isInitial, setIsInitial] = useState(true);

    useEffect(() => {
        overflow.current.style.maxHeight = "unset";
        overflow.current.style.opacity = "0";
        overflow.current.style.transition = "all 0s";

        setTimeout(() => {
            setHeight(overflow.current.offsetHeight * 1.1);
            overflow.current.style.maxHeight = !toggle ? overflow.current.offsetHeight + "px" : 0;
            overflow.current.style.padding = !toggle ? "1% 5%" : "0% 5%";
            overflow.current.style.opacity = 1;
            overflow.current.style.transition = "all 2s";
        }, 10);
    },[dimensions]);
    
    useEffect(() => {
        if(!isInitial){
            if(!isAnimating){
                overflow.current.style.maxHeight = !toggle ? height + "px" : 0;
                overflow.current.style.padding = !toggle ? "1% 5%" : "0% 5%";

                setIsAnimating(true);

                setTimeout(() => {
                    setIsAnimating(false);
                }, 2000);
            }
        }
    }, [toggle]);

    useEffect(() => {
        setIsAnimating(true);
        setToggle(true);
        setTimeout(() => {
            overflow.current.style.maxHeight = 0 + "px";
            overflow.current.style.padding = "0% 5%";

            setTimeout(() => {
                setIsAnimating(false);
                setIsInitial(false);
            }, 2000);
        }, 4000);
    }, [])

    const toggleHeight = () => {
        if(!isAnimating){
            setToggle(toggle => setToggle(!toggle));
        }
    }

    return(
        <div className={'list ' + className }> 
            <div onClick={toggleHeight}>
                <Ribon ribon={ribon}>
                    <ListSvg />
                </Ribon>
            </div>
            <div className="overflow" ref={overflow}>
                {children}
            </div>
        </div>
    )
}

export default List