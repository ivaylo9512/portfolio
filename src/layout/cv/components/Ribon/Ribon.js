import Rect from '@svgIcons/Rect';
import './Ribon.scss';

const Ribon = ({ name, ribon, children }) => {
    return(
        <div className='ribon'>
            <div className='content'>
                {ribon ? ribon : name ? <span>{name}</span> : ""}
                {children}
            </div>
            <Rect height='85' width={564}/>
        </div>
    )
}

export default Ribon