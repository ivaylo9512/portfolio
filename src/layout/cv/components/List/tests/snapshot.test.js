import List from '../List';
import { shallow } from 'enzyme';
import Ribon from '@cv/components/Ribon/Ribon';

describe('List snapshot tests', () => {
    const createWrapper = () => shallow(
        <List name='test-name' className='test-class'>
            <Ribon />
        </List>
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper();
     
        expect(wrapper).toMatchSnapshot();
    })
});