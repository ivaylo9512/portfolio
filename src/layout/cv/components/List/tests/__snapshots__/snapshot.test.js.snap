// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List snapshot tests should match snapshot 1`] = `
<div
  className="list test-class"
>
  <Ribon
    name="test-name"
  >
    <ListSvg />
  </Ribon>
  <Ribon />
</div>
`;
